src/Features/Core/Portable/Completion/Providers/AbstractContextVariableArgumentProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal abstract class AbstractContextVariableArgumentProvider : ArgumentProvider
+    {
+        public override async Task ProvideArgumentAsync(ArgumentContext context)
+        {
+            if (context.PreviousValue is not null)
+            {
+                return;
+            }
+
+            var semanticModel = await context.Document.GetRequiredSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
```

Comment:
possibly worthwhile to just have this on the context object?

Reply:
I have already moved it in a local follow-up PR.